x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/string/LocateFunctionProcessor.java

```diff
@@ -59,6 +59,11 @@ public static Integer doProcess(Object pattern, Object input, Object start) {
         if (start != null && !(start instanceof Number)) {
             throw new SqlIllegalArgumentException("A number is required; received [{}]", start);
         }
+    
+        if (start != null && ((((Number) start).longValue() > Integer.MAX_VALUE) || (((Number) start).longValue() - 1 < Integer.MIN_VALUE))) {
```

Comment:
I'd have an `if (start != null)` as an upper level if and then have inside it a check like that:
```
if (start instanceof Number == false) {
...
} else if (<range check>) {
...
}
```